### PR TITLE
ISPN-14887 Publisher prefer to stay local with shared store or REPL

### PR DESCRIPTION
**Pull request by wburns**
* Local target when shared store in use
* Reduce delivery guarantee for REPL and shared store on local node

https://issues.redhat.com/browse/ISPN-14887

### CODE REVIEW

**Review by fax4ever — COMMENTED** [review]
Very nice job. I have only one comment here.

File: core/src/main/java/org/infinispan/reactive/publisher/impl/ClusterPublisherManagerImpl.java [hunk 1]
```diff
@@ -887,13 +902,15 @@ public Long fromCacheEntry(CacheEntry entry) {
...
-   private boolean hasWriteBehindSharedStore(PersistenceConfiguration persistenceConfiguration) {
+   private void updateStoreInfo(PersistenceConfiguration persistenceConfiguration) {
       for (StoreConfiguration storeConfiguration : persistenceConfiguration.stores()) {
-         if (storeConfiguration.shared() && storeConfiguration.async().enabled()) {
-            return true;
+         if (storeConfiguration.shared()) {
+            sharedStore = true;
+            if (storeConfiguration.async().enabled()) {
+               writeBehindShared = true;
+            }
```
[the remark this answers is not in the document]
Comment (wburns) [reply core/src/main/java/org/infinispan/reactive/publisher/impl/ClusterPublisherManagerImpl.java:911]: We have never really defined what happens with more than 1 store. In actuality what happens is the first store that returns that it supports BULK_READ is used to handle iteration.

[the remark this answers is not in the document]
Comment (wburns) [reply core/src/main/java/org/infinispan/reactive/publisher/impl/ClusterPublisherManagerImpl.java:911]: Yeah, I have never seen a customer with more than 1 store configured other than with rolling upgrades and that is a temporary thing and the extra store is already shared, so it wouldn't cause an issue here.

**Comment by fax4ever** [comment]
Thanks @wburns. I merged the backport and re run the CI (there was an error on the job)

**Comment by wburns** [comment]
> CI issue is related?

Looking at the failure I can't see how it would. The cache in the test is a LOCAL cache, so it wouldn't even touch this code here.

**Comment by tristantarrant** [comment]
Merged, thanks

**Pull request merged**